packages/@aws-cdk/aws-appsync/test/appsync-grant.test.ts

```diff
@@ -0,0 +1,1031 @@
+      authorizationConfig: {
+        defaultAuthorization: {
+          authorizationType: appsync.AuthorizationType.IAM,
+        },
+      },
+    });
+
+    // WHEN
+    const grantResources = [
+      { custom: 'types/Mutation/fields/addTest' },
+    ];
+    api.grant(role, grantResources, 'appsync:graphql');
+
+    // THEN
+    expect(stack).toHaveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Action: 'appsync:graphql',
+            Effect: 'Allow',
+            Resource: {
+              'Fn::Join': [
+                '',
+                [
+                  'arn:',
+                  {
+                    Ref: 'AWS::Partition',
+                  },
+                  ':appsync:',
+                  {
+                    Ref: 'AWS::Region',
+                  },
+                  ':',
+                  {
+                    Ref: 'AWS::AccountId',
+                  },
+                  ':apis/',
+                  {
+                    'Fn::GetAtt': [
+                      'API62EA1CFF',
+                      'ApiId',
+                    ],
+                  },
+                  '/types/Mutation/fields/addTest',
+                ],
+              ],
+            },
+          },
+        ],
+      },
+    });
+  });
+
+  test('grant provides correct TYPE (no FIELD) permissions', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const role = new iam.Role(stack, 'Role', {
+      assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+    });
+    const api = new appsync.GraphQLApi(stack, 'API', {
+      name: 'demo',
+      schemaDefinitionFile: join(__dirname, 'appsync.test.graphql'),
+      authorizationConfig: {
+        defaultAuthorization: {
+          authorizationType: appsync.AuthorizationType.IAM,
+        },
+      },
+    });
+
+    // WHEN
+    const grantResources = [
+      { type: 'Mutation' },
+    ];
+    api.grant(role, grantResources, 'appsync:graphql');
+
+    // THEN
+    expect(stack).toHaveResource('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Action: 'appsync:graphql',
+            Effect: 'Allow',
+            Resource: {
+              'Fn::Join': [
+                '',
+                [
+                  'arn:',
+                  {
+                    Ref: 'AWS::Partition',
+                  },
+                  ':appsync:',
+                  {
+                    Ref: 'AWS::Region',
+                  },
+                  ':',
+                  {
+                    Ref: 'AWS::AccountId',
+                  },
+                  ':apis/',
+                  {
+                    'Fn::GetAtt': [
+                      'API62EA1CFF',
+                      'ApiId',
+                    ],
+                  },
+                  '/types/Mutation/*',
+                ],
+              ],
```

Comment:
Try to be considerate of the readers of this test, and format it to waste less vertical space:

```ts
              'Fn::Join': ['', [
                  'arn:',
                  { Ref: 'AWS::Partition', },
                  ':appsync:',
                  { Ref: 'AWS::Region' },
                  ':',
                  { Ref: 'AWS::AccountId' },
                  ':apis/',
                  { 'Fn::GetAtt': ['API62EA1CFF', 'ApiId'] },
                  '/types/Mutation/*',
                ]],
```

Same comment in all remaining tests.